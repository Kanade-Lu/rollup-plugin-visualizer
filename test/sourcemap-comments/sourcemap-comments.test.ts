import { OutputAsset, OutputChunk, OutputOptions, rollup } from "rollup";
import { describe, it, expect } from "@jest/globals";
import terser from '@rollup/plugin-terser';
import { ALL_TEMPLATE } from "../util";

// mock random for stable id
let randomCall = 0;
Math.random = () => randomCall / 100;

import { visualizer } from "../../dist/plugin";

describe("Sourcemap Comments", () => {
    const input = {
      index: "test/sourcemap-comments/input.js",
    };
  
    it("test", async () => {
      const inputOptions = {
        input,
        plugins: [
            terser(),
          visualizer({
            filename: `stats`,
            template: 'raw-data',
            emitFile: true,
            sourcemap: true,
            brotliSize: true,
            gzipSize: true
          }),
        ],
      };
      const outputOptions: OutputOptions = {
        format: "es",
        dir: "./temp/",
        sourcemap: true
      };
  
      const bundle = await rollup(inputOptions);
  
      const result = await bundle.generate(outputOptions);

      const bundleChunk = result.output.find(file => file.fileName === "index.js")

      expect(bundleChunk).not.toBeNull();
      const bundleChunkCode = (bundleChunk as OutputChunk).code
      expect(bundleChunkCode).toMatchSnapshot();
      expect(Buffer.byteLength(bundleChunkCode)).toMatchSnapshot();
  
      const generatedStats = result.output.find((file) => file.fileName === "stats");
  
      expect(generatedStats).not.toBeNull();
      expect((generatedStats as OutputAsset).source).toMatchSnapshot();
    });
  });
  