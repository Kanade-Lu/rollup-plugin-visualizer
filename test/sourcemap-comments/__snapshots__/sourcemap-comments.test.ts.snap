// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sourcemap Comments > test 1`] = `
"alert("hello");
//# sourceMappingURL=index.js.map
"
`;

exports[`Sourcemap Comments > test 2`] = `50`;

exports[`Sourcemap Comments > test 3`] = `
"{
  "version": 2,
  "tree": {
    "name": "root",
    "children": [
      {
        "name": "index.js",
        "children": [
          {
            "name": "test/sourcemap-comments/input.js",
            "uid": "38ca2e3e-1"
          }
        ]
      }
    ],
    "isRoot": true
  },
  "nodeParts": {
    "38ca2e3e-1": {
      "renderedLength": 16,
      "gzipLength": 0,
      "brotliLength": 0,
      "metaUid": "38ca2e3e-0"
    }
  },
  "nodeMetas": {
    "38ca2e3e-0": {
      "id": "/test/sourcemap-comments/input.js",
      "moduleParts": {
        "index.js": "38ca2e3e-1"
      },
      "imported": [],
      "importedBy": [],
      "isEntry": true
    }
  },
  "env": {
    "rollup": "4.29.1"
  },
  "options": {
    "gzip": false,
    "brotli": false,
    "sourcemap": true
  }
}"
`;
